### ISSUE TITLE
Async arrow functions always fail arrow-parens rule.

### ISSUE THREAD

**Issue by leebyron** [eslint/eslint#3354]
Async arrow functions always fail arrow-parens rule.

```
async () => {
}
```

Produces error:

```
error  Expected parentheses around arrow function argument     arrow-parens
```

Note: I'm using the `babel-eslint` parser in order to support the async arrow function syntax. cc @hzoo

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by hzoo** [comment]
Hm, this is probably another rule that needs to be added to https://github.com/babel/eslint-plugin-babel similar to the `generator-star-spacing` rule and asyc/await. Don't think we can do much from the babel-eslint side other than transforming the code not to be `async` while linting.

`var token = context.getFirstToken(node); // returns async instead of ( or an identifier` - unless you have another suggestion?

**Comment by nzakas** [comment]
We can't support this natively, please follow up with @hzoo.

**Comment by swarthy** [comment]
Hi!
Node.js v8+ now support native async/await and statement like
```javascript
const fn = async (getFoo, getBar) => {
  const foo = await getFoo()
  const bar = await getBar()
  return foo + bar
}
```
provides error: ```Unexpected space before function parentheses. (space-before-function-paren)```

.eslintrc contains
```
{
  rules: {
    "space-in-parens": [
        2,
        "never"
    ],
}
```
Should eslint support this case now?
